prac_10/testing.py

```diff
@@ -0,0 +1,77 @@
+"""
+CP1404/CP5632 Practical
+Testing demo using assert and doctest
+"""
+
+import doctest
+from prac_06.car import Car
+
+
+def repeat_string(s, n):
+    """Repeat string s, n times, with spaces in between."""
+    string_list = [s] * n
+    return " ".join(string_list)
+
+
+def is_long_word(word, length=5):
+    """
+    Determine if the word is as long or longer than the length passed in
+    >>> is_long_word("not")
+    False
+    >>> is_long_word("supercalifrag")
+    True
+    >>> is_long_word("Python", 6)
+    True
+    """
+    return len(word) >= length
+
+
+def create_sentence(string):
+    if not string.endswith('.'):
+        string += '.'
+    return string.capitalize()
+
+
+def run_tests():
+    """Run the tests on the functions."""
+    # assert test with no message - used to see if the function works properly
+    assert repeat_string("Python", 1) == "Python"
+    # the test below should fail
+    assert repeat_string("hi", 2) == "hi hi"
+
+    # Hint: "-".join(["yo", "yo"] -> "yo-yo"
+
+    # assert test with custom message,
+    # used to see if Car's init method sets the odometer correctly
+    # this should pass (no output)
+    test_car = Car()
+    assert test_car._odometer == 0, "Car does not set odometer correctly"
+
+    # Note that Car's __init__ function sets the fuel in one of two ways:
+    # using the value passed in or the default
+    # You should test both of these
+    assert test_car.fuel == 0
+    test_car = Car(fuel=10)
+    assert test_car.fuel == 10
+
+    assert create_sentence('hello') == 'Hello.'
+    assert create_sentence('It is an ex parrot.') == 'It is an ex parrot.'
+    assert create_sentence('The programming assignment was fun') == 'The programming assignment was fun.'
```

Comment:
This testing works but I believe that we were meant to use doctests like the is_long_word function